cluster-sync/sync-os-ci.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash -e
+
+cdi=$1
+cdi="${cdi##*/}"
+
+echo $cdi
+
+source ./hack/build/config.sh
+source ./hack/build/common.sh
+source ./cluster-up/hack/common.sh
+source ./cluster-up/cluster/${KUBEVIRT_PROVIDER}/provider.sh
+source ./cluster-sync/${KUBEVIRT_PROVIDER}/provider.sh
+
+CDI_INSTALL="install-operator"
+CDI_NAMESPACE=${CDI_NAMESPACE:-cdi}
+CDI_INSTALL_TIMEOUT=${CDI_INSTALL_TIMEOUT:-120}
+CDI_AVAILABLE_TIMEOUT=${CDI_AVAILABLE_TIMEOUT:-480}
+CDI_UPGRADE_RETRY_COUNT=${CDI_UPGRADE_RETRY_COUNT:-60}
+
+# Set controller verbosity to 3 for functional tests.
+export VERBOSITY=3
+
+PULL_POLICY=$(getTestPullPolicy)
+# The default DOCKER_PREFIX is set to kubevirt and used for builds, however we don't use that for cluster-sync
+# instead we use a local registry; so here we'll check for anything != "external"
+# wel also confuse this by swapping the setting of the DOCKER_PREFIX variable around based on it's context, for
+# build and push it's localhost, but for manifests, we sneak in a change to point a registry container on the
+# kubernetes cluster.  So, we introduced this MANIFEST_REGISTRY variable specifically to deal with that and not
+# have to refactor/rewrite any of the code that works currently.
+MANIFEST_REGISTRY=$DOCKER_PREFIX
+
+# Need to set the DOCKER_PREFIX appropriately in the call to `make docker push`, otherwise make will just pass in the default `kubevirt`
+DOCKER_PREFIX=$DOCKER_PREFIX make push
+exit 0
```

Comment:
exit 0 mid-script seems to make a lot of the lines below to be dead code

Reply:
:man_facepalming: :smile: